### DISCUSSION TITLE
[dropdown] no logic that handles long content

### DISCUSSION THREAD

**Discussion by giladv** [radix-ui/primitives#1065] [Help]
Hello,

Here's an example of what i mean:
https://codesandbox.io/s/kdbbq?module=App.js&file=/App.js

essentially, a long list of items on a dropdown stretches the entire page.

### expected behavior
The dropdown is trimmed in height to fit the entire available vertical space, added an overflow auto and maybe even autoscroll when a hidden item is focused. 
 
also, even if i wanted to implement that kind of a behavior myself, the logic popper js logic is not exposed so we can't easily calculate the content height + absolute location to know how to size it..

or am i missing something?

**Comment by benoitgrelard** [comment]
Hi @giladv, yes you are correct that this is a limitation of this primitive currently.
We are doing this in the `Select` we're currently building as it is much more needed there usually, but hopefully we can bring this kind of thing to the other menus later.

**Reply by giladv** [reply]
Thanks @benoitgrelard,
is it possible -as a temp solution- to expose some of popper js's api so it is possible for users to implement their own solution for now?

**Reply by benoitgrelard** [reply]
I don't think it would easily be done unfortunately.
